### PR TITLE
Improved contributing instructions

### PR DESCRIPTION
**Pull request by Martinsos**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,41 +1,43 @@
 Thanks so much for considering contributing to Open SaaS 🙏
 
 ## Considerations before Contributing
+Check if there is a GitHub issue already for the thing you would like to work on. If there is no issue yet, create a new one.
 
-### General Considerations
-1. If there's something you'd like to add, and the issue doesn't already exist, create a new one and assign yourself to it. Wait until we've agreed on a plan of action before beginning your work.
```
Comment (Martinsos) [inline CONTRIBUTING.md:6]: I thought telling them they can't start without getting an approval from is too harsh, so I replaced that with more of a guideline than an order.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,41 +1,43 @@
...
-2. If the issue does already exist, and noone is assigned to it, assign yourself and feel free to begin working on it.
+Let us know, in the issue, that you would like to work on it and how you plan to approach it.
+This helps, especially with the more complex issues, as it allows us to discuss the solution upfront and make sure it is well planned and fits with the rest of the project.
 
-### How Users Get the Starter Template
+## How to Contribute
```
Comment (Martinsos) [inline CONTRIBUTING.md:9]: I pushed this part up, as I think this is what they want to know, how to actually contribute, before reading the details of how we do the versioning (at least that is how it was for me).

I also made steps a bit more concrete / practical, and also mentioned e2e tests and docs.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,41 +1,43 @@
...
+2. Check out this repo (`main` branch) and make sure you are able to get the app in `app/` running (to set it up, follow the same steps as for running a new open-saas app, as explained in the open-saas docs).
+3. Create a new git branch for your work (aka feature branch) and do your changes on it.
+4. Update e2e tests in [e2e-tests](/e2e-tests/) if needed and make sure they are passing.
+5. Create a pull request (towards `main` as a base branch).
+6. If docs (also) need updating, check out the `deployed-version` branch, make your own feature branch from it, make changes in [blog/src/content/docs](/blog/src/content/docs/), and submit another PR with those changes (towards `deployed-version` as a base branch).
+7. Make a "Da Boi" meme while you wait for us to review your PR(s).
+8. If you don't know who "Da Boi" is, head back to the [Wasp Discord](https://discord.gg/aCamt5wCpS) and find out :)
 
-We currently have two ways to pull the template:
-1. the `use this template` button on the [repo homepage](https://github.com/wasp-lang/open-saas)
```
Comment (Martinsos) [inline CONTRIBUTING.md:12]: If I am correct we don't allow them to use this template directly from the repo anymore, so I dropped this bullet point.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,41 +1,43 @@
...
 
-**This means, that whenever a user pulls the template, they are getting the version present in the most recent commit on `main`**
+In the case of Open SaaS, which is a Wasp template, the tag is `wasp-v{{version}}-template`, where `{{version}}` is the current version of Wasp, e.g. `wasp-v0.13-template`.
 
-Also, If we update Wasp to a new major version, we should also update the `TAG_NAME` in the action.
+For simplicity, in Open SaaS, we automatically re-apply the tag to the most recent commit on the `main` branch via the `.github/workflows/retag-commit.yml` workflow. This way, users always get the latest version of the template (as on `main` branch) when they start a new project via `wasp new -t saas`.
+
+**This means, that whenever a user pulls the template via `wasp new -t saas`, they are getting the version present in the most recent commit on `main`.**
+
+NOTE: When Wasp releases a new major version, we should also make sure to update Open SaaS to work with this new version. In PR that will bring this update, we should also make sure to update the `TAG_NAME` in the GitHub Workflow that does the tagging, to be the template tag used by the newest version of Wasp.
```
Comment (Martinsos) [inline CONTRIBUTING.md:31]: Is this correct, this NOTE?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,41 +1,43 @@
...
 
-The default, clean template that users get when cloning the starter lives on `main`, while `deployed-version` is what you see when you go to [OpenSaaS.sh](https://opensaas.sh) and the [docs](https://docs.opensaas.sh) 
+The default, clean template that users get when cloning the starter lives on `main`, while `deployed-version` is somewhat modified version of that same template which you see when you go to [OpenSaaS.sh](https://opensaas.sh) and the [docs](https://docs.opensaas.sh).
 
-If you want to make changes to the default starter template, base feature branches and Pull Requests off of `main`
-If you want to make changes to the OpenSaaS.sh site or it's Documentation, base feature branches and Pull Requests off of `deployed-version`
+If you want to make changes to the default starter template, base feature branches and Pull Requests off of `main`.
+If you want to make changes to the OpenSaaS.sh site or it's Documentation, base feature branches and Pull Requests off of `deployed-version`.
 
-## How to contribute
```
Comment (Martinsos) [inline CONTRIBUTING.md:34]: I removed these instructions and instead we just point them directly to CONTRIBUTING.md now. So there is no duplication and it is clear where they need to go.

File: README.md [hunk 6]
```diff
@@ -63,25 +63,15 @@ For everything you need to know about getting started and using this template, c
 
 We've documented everything in great detail, including installation instructions, pulling updates to the template, guides for integrating services, SEO, deployment, and more. 🚀
 
+## Getting Help & Providing Feedback
 
-## Changes & Contributions
-
-### Template Versioning
```
Comment (Martinsos) [inline README.md:69]: I removed these instructions and instead we just point them directly to CONTRIBUTING.md now. So there is no duplication and it is clear where they need to go.

File: README.md [hunk 7]
```diff
@@ -63,25 +63,15 @@ For everything you need to know about getting started and using this template, c
...
+1. [Open an issue](https://github.com/wasp-lang/open-saas/issues)
+2. [Wasp Discord](https://discord.gg/aCamt5wCpS) -- please direct questions to the #🙋questions forum channel
 
-### Contributing
+## Contributing
 
 Note that we've tried to get as many of the core features of a SaaS app into this template as possible, but there still might be some missing features or functionality.
 
-We could always use some help tying up loose ends, so consider [contributing](https://github.com/wasp-lang/open-saas/blob/main/CONTRIBUTING.md)!
+We could always use some help tying up loose ends: contributions are welcome! Check out [CONTRIBUTING.md](/CONTRIBUTING.md) for more details.
```
Comment (Martinsos) [inline README.md:76]: I used this trick with `/...` to link to the stuff in the GitHub repo directly, without having `https://github.com/....` in front, which makes it more robust and simpler.

File: CONTRIBUTING.md (hunk 4 above)
Comment (vincanger) [reply CONTRIBUTING.md:31]: yep @Martinsos this sounds right

File: README.md (hunk 7 above)
Comment (vincanger) [reply README.md:76]: TIL

**Review by vincanger — APPROVED** [review]

**Pull request merged**